cdap-app-templates/cdap-etl/cdap-data-pipeline-base/src/main/java/io/cdap/cdap/datapipeline/service/ValidationHandler.java

```diff
@@ -91,6 +81,11 @@ public void healthCheck(HttpServiceRequest request, HttpServiceResponder respond
   @Path("v1/contexts/{context}/validations/stage")
   public void validateStage(HttpServiceRequest request, HttpServiceResponder responder,
                             @PathParam("context") String namespace) throws IOException, AccessException {
+    validateLocally(request, responder, namespace);
+  }
+
+  private void validateLocally(HttpServiceRequest request, HttpServiceResponder responder,
```

Comment:
Why do you need to split it? I don't see this reused

Reply:
There will be a config check introduced to put remote execution behind a flag. There will be validateLocally and validateRemotely methods which I thought would give more readability.